pytest-xdist fails with an error related to py.log

In PR #256 all tests on Linux failed (https://github.com/alchemistry/alchemlyb/pull/256#issuecomment-1290851926) with [an errors in pytest-xdist](https://github.com/alchemistry/alchemlyb/actions/runs/3322034917/jobs/5490506237#step:6:39)
```
Run pytest -v -n 2 --cov=alchemlyb --cov-report=xml --color=yes src/alchemlyb/tests
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/_pytest/main.py", line 266, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1037, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/pluggy/_hooks.py", line 277, in call_historic
INTERNALERROR>     res = self._hookexec(self.name, self.get_hookimpls(), kwargs, False)
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/xdist/plugin.py", line 192, in pytest_configure
INTERNALERROR>     from xdist.dsession import DSession
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/_pytest/assertion/rewrite.py", line 168, in exec_module
INTERNALERROR>     exec(co, module.__dict__)
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/xdist/dsession.py", line 5, in <module>
INTERNALERROR>     from xdist.scheduler import (
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/_pytest/assertion/rewrite.py", line 168, in exec_module
INTERNALERROR>     exec(co, module.__dict__)
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/xdist/scheduler/__init__.py", line 1, in <module>
INTERNALERROR>     from xdist.scheduler.each import EachScheduling  # noqa
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/_pytest/assertion/rewrite.py", line 168, in exec_module
INTERNALERROR>     exec(co, module.__dict__)
INTERNALERROR>   File "/home/runner/micromamba-root/envs/test/lib/python3.10/site-packages/xdist/scheduler/each.py", line 1, in <module>
INTERNALERROR>     from py.log import Producer
INTERNALERROR> ModuleNotFoundError: No module named 'py.log'; 'py' is not a package
Error: Process completed with exit code 3.
```

The version of pytest-xdist is 2.5.0. A [newer version 3.0.2 was released](https://pytest-xdist.readthedocs.io/en/latest/changelog.html#pytest-xdist-3-0-2-2022-10-25) which removes dependence on [`py.log`](https://py.readthedocs.io/en/latest/log.html) https://github.com/pytest-dev/pytest-xdist/pull/822 . 

## Solution

For right now, just disable pytest-xdist (see also https://github.com/oracle/opengrok/pull/4073) and hope that upstream fixes trickle down. Then re-enable. 


## Background

* https://github.com/pytest-dev/pytest-xdist/pull/822 — pytest-xdist removes py.log
* https://github.com/oracle/opengrok/pull/4073 — disables pytest-xdist until upstream 3.0.x is available

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -52,7 +52,7 @@ jobs:
 
     - name: Run tests
       run: |
-        pytest -v -n 2 --cov=alchemlyb --cov-report=xml --color=yes src/alchemlyb/tests
+        pytest -v --cov=alchemlyb --cov-report=xml --color=yes src/alchemlyb/tests
       env:
         MPLBACKEND: agg
 

